src/sql/workbench/services/query/common/queryModelService.ts

```
@@ -409,6 +416,25 @@ export class QueryModelService implements IQueryModelService {
 		}
 	}
 
+	public async notifyConnectionUriChanged(newUri: string, oldUri: string): Promise<void> {
+		// Get existing query runner
+		let queryRunner = this.internalGetQueryRunner(oldUri);
+		if (!queryRunner) {
+			this._logService.error(`A Query and QueryRunner was not found for '${oldUri}'`);
+		}
+		else if (this._queryInfoMap.has(newUri)) {
+			this._logService.error(`New URI '${newUri}' already has query info associated with it.`);
```

Same here, this should probably throw the error too